### PR TITLE
refactor to use closure actions

### PR DESCRIPTION
**Pull request by tchak**

### CODE REVIEW

File: app/components/todo-item/component.js [hunk 1]
```diff
@@ -14,16 +14,20 @@ export default Ember.Component.extend({
...
 
-    save() {
-      this.set('isEditing', false);
-      this.get('todo').save();
+    removeTodo(todo) {
+      todo.destroyRecord();
+    },
+
+    completeTodo(todo) {
+      todo.toggleProperty('isCompleted');
```
Comment (stefanpenner) [inline app/components/todo-item/component.js:29]: `todo.set('isCompleted', true);`

File: app/components/todos-route/component.js [hunk 2]
```diff
@@ -12,41 +11,28 @@ export default Component.extend({
   store: service(),
 
   filtered: computed('todos.@each.isCompleted', 'filter', function() {
-    var filter = this.get('filter');
-    var all = this.get('todos');
-
-    if (filter === 'all') { return all; }
-
-    return all.filterBy('isCompleted', filter === 'completed');
+    switch(this.get('filter')) {
```
Comment (stefanpenner) [inline app/components/todos-route/component.js:14]: maybe `this.get(this.get('filter'))`, or is that too clever?

File: app/components/todos-route/component.js [hunk 3]
```diff
@@ -12,41 +11,28 @@ export default Component.extend({
...
+    }
   }),
 
   completed: filterBy('todos', 'isCompleted', true),
   active: filterBy('todos', 'isCompleted', false),
+  allAreDone: computed.empty('active'),
 
-  inflection: computed('active.[]', function() {
-    var active = this.get('active.length');
+  inflection: computed('active.length', function() {
```
Comment (stefanpenner) [inline app/components/todos-route/component.js:28]: :+1:

Comment (stefanpenner) [inline app/components/todos-route/component.js:28]: another options is to use ember-inflector and one of the helpers, but I haven't been convinced that is important for such a simple example.

File: app/components/todo-item/component.js (hunk 1 above)
Comment (stefanpenner) [inline app/components/todo-item/component.js:29]: maybe we should rename the action.

File: app/components/todos-route/component.js (hunk 2 above)
Comment (tchak) [inline app/components/todos-route/component.js:14]: I'd say too clever...

Comment (stefanpenner) [inline app/components/todos-route/component.js:14]: i can accept this.

**Pull request merged**